week-1-ts-ecosystem/w01e04-light-me-up/src/light-bulb.ts

```diff
@@ -0,0 +1,43 @@
+import {PowerSource} from "./power-source";
+
+export class LightBulb {
+    protected readonly powerConsumption = 20
+    private powerSource: PowerSource | undefined;
+
+    get energyNeeded(){
+        return this.powerConsumption;
+    }
+
+    getEnergy(energy: number){
+        console.log(`I've gained ${energy}W`)
+    }
+
+    addPowerSource(powerSource: PowerSource){
+        this.powerSource = powerSource;
+    }
+
+    async lightBulbUp() {
```

Comment:
Jest Ciekawie, ale chciałbym żebyś @zajewskp nie łapał błędu tutaj tylko poza żarówką :)
Co i tak się uda
bo to jest dobrze napisane.